nodejs/dist/uws.js

```diff
@@ -209,9 +210,16 @@ class Server extends EventEmitter {
                 options.path = '/' + options.path;
             }
 
-            this.httpServer.on('upgrade', (request, socket, head) => {
-                if (!options.path || options.path == request.url.split('?')[0].split('#')[0]) {
-                    if (options.verifyClient) {
+            this._onServerUpgrade = (request, socket, head) => {
+
+                // check for wrong path
+                if (options.path) {
+                  var u = url.parse(request.url);
+                  if (u && u.pathname !== options.path)
+                    return;
```

Comment:
This exact issue has been raised at Node.js GitHub and they claim it is not supported to listen on different paths as it "makes no sense" to quote themselves. You'll have to take this up with them.